### ISSUE TITLE
Compatibility issue with MATLAB HDF5

### ISSUE THREAD

**Issue by ShuhuaGao** [JuliaIO/JLD2.jl#386]
Since JLD2 is a "HDF5-compatible file format in pure Julia", I thought the generated file can be consumed by MATLAB HDF5 routines, at least when there are only primitive types. 
```julia
using JLD2
save("./data/mydata.jld2", "int", [1, 2, 3], "f64", [1.0, 2.0])
```
Then, I try to display information of the above file with [h5disp](https://www.mathworks.com/help/matlab/ref/h5disp.html) in MATLAB. 
```matlab
>> h5disp("./data/mydata.jld2")
Warning: Unable to read '/f64' from the file. A portion of the file may be corrupt. 
> In h5info (line 108)
  In h5disp>display_hdf5 (line 121)
  In h5disp (line 99) 
Warning: Unable to read '/int' from the file. A portion of the file may be corrupt. 
> In h5info (line 108)
  In h5disp>display_hdf5 (line 121)
  In h5disp (line 99) 
HDF5 mydata.jld2 
Group '/' 
```
Other MATLAB functions like `h5read` does not work either.

However, if I did the same but with `HDF5.jl`, then MATLAB works normally. 

In short, is the file produced by JLD2 compatible with other HDF5 libraries?

**Comment by JonasIsensee** [comment]
Hi @ShuhuaGao,

for a very long time MATLAB used an old version of HDF5 `v1.8.x`.
`JLD2` is compatible with HDF5 version `> v1.10` (first released in 2016).

Please make sure, if you can update your MATLAB installation.
For me, it works just fine:

```
>> h5disp("mydata.jld2")
HDF5 mydata.jld2 
Group '/' 
    Dataset 'f64' 
        Size:  2
        MaxSize:  2
        Datatype:   H5T_IEEE_F64LE (double)
        ChunkSize:  []
        Filters:  none
        FillValue:  0.000000
    Dataset 'int' 
        Size:  3
        MaxSize:  3
        Datatype:   H5T_STD_I64LE (int64)
        ChunkSize:  []
        Filters:  none
        FillValue:  0

>> h5read("mydata.jld2", "/f64")

ans =

     1
     2

>> h5read("mydata.jld2", "/int")

ans =

  3×1 int64 column vector

   1
   2
   3

>> 
```

**Comment by ShuhuaGao** [comment]
Thanks, @JonasIsensee. I tested with MATLAB R2020a on Windows 10. How can I check the version of HDF5 used by MATLAB?

**Comment by JonasIsensee** [comment]
I've never worked with matlab, but a quick search on the internet revealed:

https://de.mathworks.com/help/matlab/ref/libraryh5.html?searchHighlight=hdf5%20version&s_tid=srchtitle_hdf5%20version_7

`[major, minor, relnum] = H5.get_libversion()`

**Comment by ShuhuaGao** [comment]
Thanks. Unfortunately, MATLAB R2020a on Windows still uses the 1.8 series of HDF5.
```
>> [major, minor, relnum] = H5.get_libversion()
major =
     1
minor =
     8
relnum =
    12
```
That explains the compatibility issue. 

Besides, the latest `JLD` package v0.13.1 plays well with MATLAB R2020a. Maybe it uses an older version of HDF5.

**Comment by JonasIsensee** [comment]
> Besides, the latest JLD package v0.13.1 plays well with MATLAB R2020a. Maybe it uses an older version of HDF5.

It's actually quite the opposite:
HDF5 has been receiving new features with every release, but for backwards compatibility they never use any of these by default. That is why JLD works in this case.
